planning/obstacle_avoidance_planner/src/node.cpp

```diff
@@ -443,7 +449,17 @@ void ObstacleAvoidancePlanner::insertZeroVelocityOutsideDrivableArea(
 
   if (enable_outside_drivable_area_stop_) {
     if (first_outside_idx) {
-      for (size_t i = *first_outside_idx; i < optimized_traj_points.size(); ++i) {
+      auto stop_idx = *first_outside_idx;
+      const auto dist = tier4_autoware_utils::calcDistance2d(
+        optimized_traj_points.at(0), optimized_traj_points.at(stop_idx));
+      const auto dist_with_margin = dist - vehicle_stop_margin_outside_drivable_area_;
+      const auto first_outside_idx_with_margin =
+        motion_utils::insertStopPoint(dist_with_margin, optimized_traj_points);
+      if (first_outside_idx_with_margin) {
+        stop_idx = *first_outside_idx_with_margin;
+      }
+      publishVirtualWall(optimized_traj_points.at(stop_idx).pose);
+      for (size_t i = stop_idx; i < optimized_traj_points.size(); ++i) {
```

Comment:
Could you make variables private like this.
```cpp
      const auto stop_idx = [&]() {
        const auto dist = tier4_autoware_utils::calcDistance2d(
          optimized_traj_points.at(0), optimized_traj_points.at(*first_outside_idx));
        const auto dist_with_margin = dist - vehicle_stop_margin_outside_drivable_area_;
        const auto first_outside_idx_with_margin =
          motion_utils::insertStopPoint(dist_with_margin, optimized_traj_points);
        if (first_outside_idx_with_margin) {
          return *first_outside_idx_with_margin;
        }
        return *first_outside_idx;
      }();

      publishVirtualWall(optimized_traj_points.at(stop_idx).pose);
      for (size_t i = stop_idx; i < optimized_traj_points.size(); ++i) {
```

Reply:
Also, even if `enable_outside_drivable_area_stop_` is false, we wanna call `publishVirtualWall`.
So the following would be better.
```cpp
if (first_outside_idx) {
    const auto stop_idx = ...;
    publishVirtualWall(optimized_traj_points.at(stop_idx).pose);
    if (enable_outside_drivable_area_stop_) {
        for (size_t i = stop_idx; i < optimized_traj_points.size(); ++i) {
            ...
        }
    }
}
```

Reply:
Thank you so much @takayuki5168 for the valuable comments :+1: .
I have updated the code, please have a look.